### PR TITLE
Don't render const keyword on stable

### LINKED ISSUE #55246: const fn shown in docs when not available
Regression from 1.26.0 to 1.27.0 and later. See for example
https://doc.rust-lang.org/1.26.0/std/vec/struct.Vec.html#method.new
https://doc.rust-lang.org/1.27.0/std/vec/struct.Vec.html#method.new
See also #31329 and #31098

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #55246.

Continuation of #55327.

r? @oli-obk

### CODE REVIEW

**Comment by oli-obk** [comment]
So I think I found the issue. The `visit_fn` function just takes the plain `FnHeader` from the `HIR`, which contains `Constness` irrelevant of stability.

Etiher the `visit_fn` method or the `Clean` impl for `doctree::Function` need to access the `tcx` to filter out the constness instead of just passing it down.

File: src/librustdoc/clean/mod.rs [hunk 1]
```diff
@@ -1674,18 +1674,29 @@ impl Clean<Item> for doctree::Function {
...
             visibility: self.vis.clean(cx),
             stability: self.stab.clean(cx),
             deprecation: self.depr.clean(cx),
-            def_id: cx.tcx.hir().local_def_id(self.id),
+            def_id: did,
             inner: FunctionItem(Function {
                 decl,
                 generics,
-                header: self.header,
+                header: hir::FnHeader {
```
Comment (oli-obk) [inline src/librustdoc/clean/mod.rs]: you can use struct update syntax for this: `hir::FnHeader { constness, ..self.header }`

**Review by oli-obk — CHANGES_REQUESTED** [review]
r=me with nix fixed

**Comment by GuillaumeGomez** [comment]
Updated.

**Review by oli-obk — APPROVED** [review]

**Pull request merged**